ui/dashboards/src/components/PanelDrawer/PanelDrawer.tsx

```diff
@@ -11,242 +11,70 @@
-              value={panelName}
-              variant="outlined"
-              onChange={handlePanelNameChange}
-            />
-          </Grid>
-          <Grid item xs={4}>
-            <FormControl>
-              <InputLabel id="select-group">Group</InputLabel>
-              <Select required labelId="select-group" label="Group" value={group ?? 0} onChange={handleGroupChange}>
-                {layouts.map((layout, index) => (
-                  <MenuItem key={index} value={index}>
-                    {layout.spec.display?.title || `Group ${index + 1}`}
-                  </MenuItem>
-                ))}
-              </Select>
-            </FormControl>
-          </Grid>
-          <Grid item xs={8}>
-            <TextField
-              label="Panel Description"
-              value={panelDescription}
-              variant="outlined"
-              onChange={handlePanelDescriptionChange}
-            />
-          </Grid>
-          <Grid item xs={4}>
-            <FormControl>
-              <InputLabel id="panel-type-label">Panel Type</InputLabel>
-              <PanelTypeSelect
-                required
-                labelId="panel-type-label"
-                label="Panel Type"
-                value={kind}
-                onChange={handleKindChange}
-              />
-            </FormControl>
-          </Grid>
-          <Grid item xs={12}>
-            <ErrorBoundary FallbackComponent={ErrorAlert}>
-              {OptionsEditorComponent !== undefined && options !== undefined && (
-                <OptionsEditorComponent value={options} onChange={handleOptionsChange} />
-              )}
-            </ErrorBoundary>
-          </Grid>
-        </Grid>
-      </form>
+    <Drawer isOpen={isOpen} onClose={handleClose} SlideProps={{ onExited: handleExited }}>
+      {/* When the drawer is opened, we should have a model (this also ensures the form state gets reset between opens) */}
+      {model !== undefined && (
+        <>
+          <Box
+            sx={{
+              display: 'flex',
+              alignItems: 'center',
+              marginBottom: (theme) => theme.spacing(2),
+              paddingBottom: (theme) => theme.spacing(2),
+              borderBottom: (theme) => `1px solid ${theme.palette.grey[100]}`,
+            }}
+          >
+            <Typography variant="h2">{model.drawerTitle}</Typography>
+            <Stack direction="row" spacing={1} sx={{ marginLeft: 'auto' }}>
+              {/* Using the 'form' attribute lets us have a submit button like this outside the form element */}
+              <Button type="submit" variant="contained" form={panelEditorFormId}>
+                {model.submitButtonText}
+              </Button>
+              <Button variant="outlined" onClick={handleClose}>
+                Cancel
+              </Button>
+            </Stack>
+          </Box>
+          <PanelEditorForm onSubmit={handleSubmit} initialValues={model.initialValues} />
```

Comment:
See the comment above, but the form state is now in this component which gets unmounted when the drawer is closed, allow the form state values to be reset.